### DISCUSSION TITLE
Regarding "airflow.providers.google.cloud.secrets.secret_manager.CloudSecretManagerBackend" in GCP

### DISCUSSION THREAD

**Discussion by vamsee-web** [apache/airflow#35155] [Q&A]
We started using airflow-connections-xxx in secret manager and started getting Erros related to "Secret Manager APIAccessSecretVersion airflow-variables-xxx".

The configuration is
ariflow-connections-xxx are pointed to secret manager using CloudSecretManagerBancked and
ariflow-variables-xxx should be used from the ariflow local varibles itself.

Can someone share any workaround to remove the errors related to ariflow-variables-xxx

**Comment by potiuk** [comment]
You should write your own custom backend deriving from your chosen secret and ignore variable requests (I believe returning None but you will have to check in the implementation):   https://airflow.apache.org/docs/apache-airflow/stable/security/secrets/secrets-backend/index.html#roll-your-own-secrets-backend

This is how you change behaviour of existing managers in Airflow, rolling your own secrets manager if you want different behaviour is the first-class citizen of Airflow.

**Reply by vamsee-web** [reply]
Thank you so much, i will check this is workaround

**Comment by vamsee-web** [comment]
@potiuk 
we are using Google Cloud Secret Manager Backend for airflow. I tried implementing.

[secrets]
backend_kwargs = {"variables_prefix": "null"}
backend = airflow.providers.google.cloud.secrets.secret_manager.CloudSecretManagerBackend

but this dint work. error in secret manger.
message: "Secret [projects/***/secrets/null-XXX] not found or has no versions."

**Reply by potiuk** [reply]
No idea. You need to debug it/
